lib/uvobot.rb

```diff
@@ -8,19 +8,19 @@ def run(release_date)
     if @scraper.issue_ready?(release_date)
       notify_announcements(release_date)
     else
-      @notifier.new_issue_not_published
+      @notifiers.each { |n| n.send(:new_issue_not_published) }
     end
   end
 
   private
 
   def notify_announcements(release_date)
-    page_info, announcements = @scraper.get_announcements(release_date)
+    page_info, announcements = @scraper.get_full_announcements(release_date)
```

Comment:
Nj, povodne to bolo rozdelene prave kvoli tomu aby detail parsing nevyradil vsetko. 

Notifier by potom bral v konstruktore aj scraper a dotahoval by si detail. Co je z funkcneho hladiska OK, akurat mi to uplne nesedi co tyka SRP.

Keby sme mu posunuli scraper spolu s ostatnymi datami  ako nejaky proc object predchystany. Dotahovanie detailu by bolo potom lazy.